crates/network/src/any/mod.rs

```diff
@@ -46,11 +68,537 @@ impl From<TxType> for AnyTxType {
+    fn from(value: AnyTypedTransaction) -> Self {
+        match value {
+            AnyTypedTransaction::Ethereum(tx) => Self::new(tx.into()),
+            AnyTypedTransaction::Other { ty, mut fields, .. } => {
+                fields.insert("type".to_string(), serde_json::Value::Number(ty.0.into()));
+                Self { inner: Default::default(), other: OtherFields::new(fields) }
+            }
+        }
+    }
+}
+
+impl From<AnyTxEnvelope> for WithOtherFields<TransactionRequest> {
+    fn from(value: AnyTxEnvelope) -> Self {
+        AnyTypedTransaction::from(value).into()
+    }
+}
+
+impl alloy_consensus::Transaction for AnyTypedTransaction {
+    fn chain_id(&self) -> Option<alloy_primitives::ChainId> {
+        match self {
+            Self::Ethereum(inner) => inner.chain_id(),
+            Self::Other { fields, .. } => fields.get("chainId").and_then(|v| v.as_u64()),
+        }
+    }
+
+    fn nonce(&self) -> u64 {
+        match self {
+            Self::Ethereum(inner) => inner.nonce(),
+            Self::Other { fields, .. } => {
+                fields.get("nonce").and_then(|v| v.as_u64()).expect("missing nonce in tx response")
+            }
+        }
+    }
+
+    fn gas_limit(&self) -> u64 {
+        match self {
+            Self::Ethereum(inner) => inner.gas_limit(),
+            Self::Other { fields, .. } => {
+                fields.get("gas").and_then(|v| v.as_u64()).expect("missing gas in tx response")
+            }
+        }
+    }
+
+    fn gas_price(&self) -> Option<u128> {
+        match self {
+            Self::Ethereum(inner) => inner.gas_price(),
+            Self::Other { fields, .. } => {
+                fields.get("gasPrice").and_then(|v| v.as_u64()).map(|v| v as u128)
+            }
+        }
+    }
+
+    fn max_fee_per_gas(&self) -> u128 {
+        match self {
+            Self::Ethereum(inner) => inner.max_fee_per_gas(),
+            Self::Other { fields, .. } => fields
+                .get("maxFeePerGas")
+                .and_then(|v| v.as_u64())
+                .expect("missing maxFeePerGas in tx response")
+                as u128,
+        }
+    }
+
+    fn max_priority_fee_per_gas(&self) -> Option<u128> {
+        match self {
+            Self::Ethereum(inner) => inner.max_priority_fee_per_gas(),
+            Self::Other { fields, .. } => {
+                fields.get("maxPriorityFeePerGas").and_then(|v| v.as_u64()).map(|v| v as u128)
+            }
+        }
+    }
+
+    fn max_fee_per_blob_gas(&self) -> Option<u128> {
+        match self {
+            Self::Ethereum(inner) => inner.max_fee_per_blob_gas(),
+            Self::Other { fields, .. } => {
+                fields.get("maxFeePerBlobGas").and_then(|v| v.as_u64()).map(|v| v as u128)
+            }
+        }
+    }
+
+    fn priority_fee_or_price(&self) -> u128 {
+        self.max_priority_fee_per_gas()
+            .or_else(|| self.gas_price())
+            .expect("missing maxPriorityFeePerGas or gasPrice in tx response")
```

Comment:
panicking on missing fee fields also feels wrong,
e.g. this would also panic on deposittxs

Reply:
ref https://github.com/alloy-rs/op-alloy/blob/b60f6430a9644d93702aba9de1c783366e1aac84/crates/consensus/src/transaction/deposit.rs#L184-L184

Reply:
no more expects at all 👍